src/osd/ReplicatedPG.cc

```diff
@@ -5533,7 +5533,7 @@ int ReplicatedPG::do_osd_ops(OpContext *ctx, vector<OSDOp>& ops)
 	tracepoint(osd, do_osd_op_pre_omapgetkeys, soid.oid.name.c_str(), soid.snap.val, start_after.c_str(), max_return);
 	set<string> out_set;
 
-	if (pool.info.supports_omap()) {
+	if (obs.oi.is_omap()) {
```

Comment:
and there are two "Signed-off-by: Jianpeng Ma <jianpeng.ma@intel.com>" in your commit message, could you remove one of them?